app/src/pages/SingleContent.vue

```diff
@@ -34,6 +38,32 @@ watch(
     { immediate: true },
 );
 
+watch(
+    appLanguageAsRef,
+    async () => {
+        if (!content.value) {
+            return;
+        }
+
+        if (appLanguageAsRef.value?._id != content.value.language) {
+            const contentDocs = await db.whereParent(content.value.parentId);
+            const preferred = contentDocs.find((c) => c.language == appLanguageAsRef.value?._id);
+
+            if (preferred) {
+                content.value = preferred;
+                await router.replace({ name: "content", params: { slug: preferred.slug } });
+                return;
+            }
+            useNotificationStore().addNotification({
```

Comment:
I would still like a test for this notification. Did you look at that? Just test switching to a language for which there is no translation and then test if `addNotification` was not called:

```ts
 expect(notificationStore.addNotification).not.toHaveBeenCalled()
```

Reply:
@belljohan3 I see I didn't mention it in this comment but a test that the notification _was_ called is also good I think.

Reply:
This is still missing, we only test that when the language does change the notification also is displayed, please also add a test that when the language doesn't change because it isn't available the notification is displayed.